justfile

```
@@ -138,3 +138,99 @@ update-ffi-flake: check-nix
 
     echo "checking for a consistent golang verion"
     ../scripts/run-just.sh check-golang-version
+
+# Trigger Reexecution Benchmark
+# Usage: just benchmark-trigger <firewood> <avalanchego> <task> <runner> [libevm]
+benchmark-trigger firewood avalanchego task runner libevm="": check-nix
+    #!/usr/bin/env bash
+    set -euo pipefail
+    
+    GH="nix run ./ffi#gh --"
+    
+    LIBEVM_FLAG=""
+    if [ -n "{{ libevm }}" ]; then
+        LIBEVM_FLAG="-f libevm={{ libevm }}"
+    fi
+    
+    $GH workflow run "Firewood Reexecution Benchmark" \
+        --repo ava-labs/avalanchego \
+        --ref "{{ avalanchego }}" \
+        -f firewood="{{ firewood }}" \
+        -f task="{{ task }}" \
+        -f runner="{{ runner }}" \
+        $LIBEVM_FLAG
+    
+    sleep 10
+    
+    RUN_ID=$($GH run list \
+        --repo ava-labs/avalanchego \
+        --workflow "Firewood Reexecution Benchmark" \
+        --limit 5 \
+        --json databaseId,createdAt \
+        --jq '[.[] | select(.createdAt | fromdateiso8601 > (now - 60))] | .[0].databaseId')
+    
+    if [ -z "$RUN_ID" ] || [ "$RUN_ID" = "null" ]; then
+        echo "Error: Could not find triggered workflow run" >&2
+        exit 1
+    fi
+    
+    echo "$RUN_ID"
+
+# Wait for reexecution benchmark run to complete
+# Usage: just benchmark-wait <run_id>
+benchmark-wait run_id: check-nix
+    #!/usr/bin/env bash
+    set -euo pipefail
+    nix run ./ffi#gh -- run watch "{{ run_id }}" --repo ava-labs/avalanchego --exit-status
```

This can hang if it never finishes, which is possible if someone introduces some infinite loop somewhere.

I think github will let it run for a day, which is way too long. 2h seems like plenty but perhaps a shorter time makes sense.

```suggestion
    nix run ./ffi#gh -- timeout -k 1m 2h run watch "{{ run_id }}" --repo ava-labs/avalanchego --exit-status
```